src/App.tsx

```diff
@@ -12,6 +13,18 @@ export function App() {
     setGreetMsg(await invoke("greet", { name }));
   }
 
+  const { data } = useQuery({
+    queryKey: ["get_commits"],
+    queryFn: async () => {
+      return await invoke("get_commits");
+    },
+  });
+
+  async function getCommitInfo(hash: string) {
+    const commitInfo = await invoke("get_commit", { hash });
```

Comment:
Use `react-query` `useMutation` for it or `useQuery` with state variable for selected commit